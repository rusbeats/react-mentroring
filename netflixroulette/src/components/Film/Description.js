import React from 'react';
// Hardcoding the value only for homework #3
const filmDescription = 'The Matrix is a 1999 science fiction action'
                + ' film written and directed by The Wachowskis and'
                 + ' starring Keanu Reeves, Laurence Fishburne, Carrie-Anne Moss,'
                  + ' Hugo Weaving, and Joe Pantoliano. It depicts a dystopian future'
                  + ' in which humanity is unknowingly trapped inside a simulated reality'
                   + ' called the Matrix, created by thought-capable machines (artificial beings)'
                    + ' to control humans while using their bodies as an energy source.';

const Description = () => (
  <div className="film-desc">
    {filmDescription}
  </div>
);
export default Description;
