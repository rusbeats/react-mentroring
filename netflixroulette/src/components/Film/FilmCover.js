/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable react/destructuring-assignment */
import React from 'react';

const FilmCover = props => (
  <div className="film-cover">
    <img className="film-image" src={props.filmImage} />
  </div>
);
export default FilmCover;
