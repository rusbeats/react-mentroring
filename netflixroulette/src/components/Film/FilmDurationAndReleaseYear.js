import React from 'react';
// Hardcoding the value only for homework #3
const duration = '152 min';
const releaseYear = '2002';

const FilmDurationAndReleaseYear = () =>(
            <div className="film-duration-and-release-year">
                <span>{duration} &nbsp; {releaseYear}</span>
            </div>
    );
export default FilmDurationAndReleaseYear;

