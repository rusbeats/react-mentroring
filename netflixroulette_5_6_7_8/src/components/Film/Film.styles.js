import constants from '../../utils/constants';
import styled from 'styled-components';

const { colors } = constants;

export const FilmTitleAndDateWrapper = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 10px 0;
`;

export const FilmTitle = styled.div`
    text-transform: uppercase;
    font-size: 20px;
    font-weight: bold;
    max-width: 250px;
`;

export const FilmDate = styled.div`
    border-style: solid;
    border-width: 1.5px;
    border-color: ${colors.grey};
    border-radius: 3px;
    padding: 3px 10px;
`;

export const FilmGenres = styled.div`
    color: ${colors.grey};
    max-width: 350px;
`;
