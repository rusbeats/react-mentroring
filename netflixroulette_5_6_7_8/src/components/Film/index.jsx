import React from 'react';
import PropTypes from 'prop-types';
import { getYear } from '../../utils/helpers';
import { Image } from '../Image/Image.styles';
import { FilmTitleAndDateWrapper, FilmTitle, FilmDate, FilmGenres } from './Film.styles';

const Film = ({ film, onClick }) => {
    const releaseYear = getYear(film.release_date);
    const genresLength = film.genres.length;

    return (
        <div onClick={onClick}>
            <Image src={film.poster_path} />
            <FilmTitleAndDateWrapper>
                <FilmTitle>{film.title}</FilmTitle>
                <FilmDate>{releaseYear}</FilmDate>
            </FilmTitleAndDateWrapper>
            <FilmGenres>
                {film.genres.map((genre, index) => (index === genresLength - 1 ? genre : genre + ' & '))}
            </FilmGenres>
        </div>
    );
};

Film.propTypes = {
    film: PropTypes.object,
    onClick: PropTypes.func
};

export default Film;
