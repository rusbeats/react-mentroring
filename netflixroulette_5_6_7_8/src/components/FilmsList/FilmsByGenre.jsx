import React from 'react';
import PropTypes from 'prop-types';
import Loader from 'react-loader-spinner';
import Film from '../Film';
import { connect } from 'react-redux';
import { getFilmFromState } from '../../store/actions';
import { StyledLink } from '../Link/Link.styles';
import { StyledFilms, StyledLoader } from './Base.styles';
import constants from '../../utils/constants';
import Link from 'next/link';
import { getFilmsByGenreState } from '../../selectors';

const { colors } = constants;

export const FilmsByGenre = ({ isFetching, getFilm, films, notFound }) => {
    if (notFound) {
        return null;
    }

    if (isFetching) {
        return (
            <StyledLoader>
                <Loader type="Puff" color="#00BFFF" height="300" width="300" />;
            </StyledLoader>
        );
    }

    const handleGetFilm = film => () => getFilm(film);

    return (
        <StyledFilms>
            {films.map(film => (
                <Link href={`/film?id=${film.id}`} as={`/film/${film.id}`} key={film.id}>
                    <StyledLink color={colors.black}>
                        <Film film={film} onClick={handleGetFilm(film)} />
                    </StyledLink>
                </Link>
            ))}
        </StyledFilms>
    );
};

StyledFilms.displayName = 'FilmsByGenre';

FilmsByGenre.propTypes = {
    isFetching: PropTypes.bool,
    getFilm: PropTypes.func,
    films: PropTypes.array,
    notFound: PropTypes.bool
};

const mapStateToProps = state => getFilmsByGenreState(state);

const mapDispatchToProps = dispatch => ({
    getFilm: film => dispatch(getFilmFromState(film))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(FilmsByGenre);
