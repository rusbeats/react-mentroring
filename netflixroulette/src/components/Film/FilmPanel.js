import React from 'react';
import FilmTitle from './FilmTitle';
import FilmCover from './FilmCover';
import FilmDurationAndReleaseYear from './FilmDurationAndReleaseYear';
import FilmRating from './FilmRating';
import Description from './Description';
import SearchButton from './SearchButton';

const FilmPanel = () => (
  <div className="film-panel">
    <FilmCover filmImage="../../src/img/matrix.jpg" />
    <div className="film-info">
      <div className="film-title-and-rating">
        <FilmTitle />
        {' '}
        <FilmRating />
      </div>
      <FilmDurationAndReleaseYear filmDuration="152 min" filmReleaseYear="2002" />
      <Description />
    </div>
    <SearchButton />
  </div>
);
export default FilmPanel;
