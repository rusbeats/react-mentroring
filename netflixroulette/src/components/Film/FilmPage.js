import React from 'react';
import FilmPanel from './FilmPanel';
import FilmByTheSameGenre from '../Item/FilmByTheSameGenre';

const FilmPage = () => (
  <div className="film-page">
    <FilmPanel />
    <FilmByTheSameGenre />
  </div>
);
export default FilmPage;
