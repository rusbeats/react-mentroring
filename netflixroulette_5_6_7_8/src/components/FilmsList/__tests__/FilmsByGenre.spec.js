import React from 'react';
import { shallow } from 'enzyme';
import { FilmsByGenre } from '../FilmsByGenre';
import Film from '../../Film';

const INITIAL_STATE = [
    {
        id: 1,
        title: 'Test title 1',
        release_date: '2000-01-31',
        poster_path: 'https://some-image-1.jpg',
        genres: ['Genre 1']
    },
    {
        id: 2,
        title: 'Test title 2',
        release_date: '2001-03-31',
        poster_path: 'https://some-image-2.jpg',
        genres: ['Genre 1', 'Genre 2']
    },
    {
        id: 3,
        title: 'Untitled Avengers',
        release_date: '2002-05-31',
        poster_path: 'https://some-image-3.jpg',
        genres: ['Genre 1', 'Genre 2', 'Genre 3']
    }
];

const PROPS = {
    film: {
        id: 1,
        genres: ['genre1', 'genre2', 'genre3'],
        runtime: 100
    },
    isFetching: true,
    notFound: true,
    getFilm: jest.fn()
};

describe('FilmsByGenre', () => {
    it('should render correctly (snapshot)', () => {
        const component = shallow(<FilmsByGenre films={INITIAL_STATE} filmFromState={PROPS.film} />);
        expect(component).toMatchSnapshot();
    });

    it('should render loader (snapshot)', () => {
        const component = shallow(<FilmsByGenre isFetching={PROPS.isFetching} filmFromState={PROPS.film} />);
        expect(component).toMatchSnapshot();
    });

    it('should not render', () => {
        const component = shallow(<FilmsByGenre notFound={PROPS.notFound} filmFromState={PROPS.film} />);
        expect(component.type()).toBeNull();
    });

    it('should handle getFilm', () => {
        const component = shallow(
            <FilmsByGenre films={INITIAL_STATE} filmFromState={PROPS.film} getFilm={PROPS.getFilm} />
        );
        component
            .find(Film)
            .at(1)
            .simulate('click');
        expect(PROPS.getFilm).toHaveBeenCalled();
    });
});
