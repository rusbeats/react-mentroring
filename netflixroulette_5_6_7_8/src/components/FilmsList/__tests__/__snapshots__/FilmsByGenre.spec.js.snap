// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsByGenre should render correctly (snapshot) 1`] = `
<FilmsByGenre>
  <Link
    as="/film/1"
    href="/film?id=1"
    key="1"
  >
    <Linkstyles__StyledLink
      color="#010101"
    >
      <Film
        film={
          Object {
            "genres": Array [
              "Genre 1",
            ],
            "id": 1,
            "poster_path": "https://some-image-1.jpg",
            "release_date": "2000-01-31",
            "title": "Test title 1",
          }
        }
        onClick={[Function]}
      />
    </Linkstyles__StyledLink>
  </Link>
  <Link
    as="/film/2"
    href="/film?id=2"
    key="2"
  >
    <Linkstyles__StyledLink
      color="#010101"
    >
      <Film
        film={
          Object {
            "genres": Array [
              "Genre 1",
              "Genre 2",
            ],
            "id": 2,
            "poster_path": "https://some-image-2.jpg",
            "release_date": "2001-03-31",
            "title": "Test title 2",
          }
        }
        onClick={[Function]}
      />
    </Linkstyles__StyledLink>
  </Link>
  <Link
    as="/film/3"
    href="/film?id=3"
    key="3"
  >
    <Linkstyles__StyledLink
      color="#010101"
    >
      <Film
        film={
          Object {
            "genres": Array [
              "Genre 1",
              "Genre 2",
              "Genre 3",
            ],
            "id": 3,
            "poster_path": "https://some-image-3.jpg",
            "release_date": "2002-05-31",
            "title": "Untitled Avengers",
          }
        }
        onClick={[Function]}
      />
    </Linkstyles__StyledLink>
  </Link>
</FilmsByGenre>
`;

exports[`FilmsByGenre should render loader (snapshot) 1`] = `
<Basestyles__StyledLoader>
  <Loader
    color="#00BFFF"
    height="300"
    type="Puff"
    width="300"
  />
  ;
</Basestyles__StyledLoader>
`;
